src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -15,10 +17,15 @@ namespace Microsoft.AspNetCore.Mvc
     /// <summary>
     /// Provides programmatic configuration for the MVC framework.
     /// </summary>
-    public class MvcOptions
+    public class MvcOptions : IEnumerable<ICompatibilitySwitch>
     {
         private int _maxModelStateErrors = ModelStateDictionary.DefaultMaxAllowedErrors;
 
+        private readonly CompatibilitySwitch<bool> _allowBindingUndefinedValueToEnumType;
+        private readonly CompatibilitySwitch<InputFormatterExceptionModelStatePolicy> _inputFormatterExceptionModelStatePolicy;
+        private readonly CompatibilitySwitch<bool> _suppressJsonDeserializationExceptionMessagesInModelState;
+        private readonly ICompatibilitySwitch[] _switches;
```

Comment:
Two of the new properties we introduced don't follow the guidelines I wrote up. We need to consider changing these to be more conservative.